powersimdata/network/europe_tub/model.py

```diff
@@ -13,61 +14,25 @@
 pypsa = _check_import("pypsa")
 
 
-class TUB:
-    """PyPSA Europe network.
+class PyPSABase(FromPyPSA):
+    """Arbitrary PyPSA network.
 
     :param str/iterable interconnect: interconnect name(s).
-    :param str zenodo_record_id: the zenodo record id. If set to None, v0.6.1 will
-        be used. If set to latest, the latest version will be used.
-    :param int reduction: reduction parameter (number of nodes in network). If None,
-        the full network is loaded.
+    :param pypsa.Network network: a PyPSA network object
     """
 
-    def __init__(self, interconnect, zenodo_record_id=None, reduction=None):
+    def __init__(self, interconnect, network):
         """Constructor."""
+        super().__init__()
         self.grid_model = "europe_tub"
```

Comment:
if it is the base class for any PyPSA network, I would say we need a `grid_model` parameter in the `__init__` that will be passed in the child class